frontend/src/components/Election/Voting/GenericBallotView.js

```diff
@@ -144,26 +142,56 @@ const Rows = ({ candidates, scores, onClick, columns }) =>
         onClick={(score) => onClick(n, score)}
         columns={columns}
       />
-      <Divider style={{ width: '100%' }} />
+      <Divider className="rowDivider"/>
     </>
   ));
 
 // Represents the list of column headings for all possible scores
-const ColumnHeadings = ({columns}) => (
+const ColumnHeadings = ({starHeadings, columns, leftTitle, rightTitle, headingPrefix}) => (
   <>
-    <Grid container alignItems="stretch">
+  { leftTitle != '' &&
+    <Grid container alignItems="stretch" >
       <Grid item xs={5}></Grid>
-      <ScoreColumnHeadings columns={columns}/>
+      <Typography align='center' className="columnDescriptor">
+        {leftTitle}
+      </Typography>
+      <Grid item xs={columns.length-2}></Grid>
+      <Typography align='center' className="columnDescriptor">
+        {rightTitle}
+      </Typography>
+    </Grid>
+  }
+    <Grid container alignItems="stretch">
+      <Grid item xs={5}>
+        { headingPrefix != '' &&
+          <Typography align='right' className="columnDescriptor">
+            {headingPrefix}
+          </Typography>
+        }
+      </Grid>
+      {columns.length > 1 &&
+        <ScoreColumnHeadings starHeadings={starHeadings} columns={columns}/>
+      }
     </Grid>
   </>
 );
 
-const ScoreColumnHeadings = ({columns}) =>
+const ScoreIcon = ({color, value}) => (
+  <div align='center'>
+    <FaStar style={{color: color, fontSize: '2.1em', transform:'translate(0%,15%)'}}/>
+    <Typography style={{position: 'absolute', transform:'translate(155%,-110%)'}} variant="h6" component="h6" className="scoreColumnHeading">
```

Comment:
We should try to make this more responsive on smaller screens. The .circle css used for the ballot bubbles might be a good reference. 

https://user-images.githubusercontent.com/41272412/221441804-984d19b1-c42d-46e9-be03-b624f40fab99.mp4

Reply:
Good catch!

.circle seems different since it's manipulating the border to be a circle, whereas for the star I'm trying to render multiple elements on top of each other

I made a change that I think fixes it? That said css is still kind of black magic to me so I just keep trying random things until it works 😅